python/ray/train/xgboost/config.py

```diff
@@ -37,28 +41,93 @@ class XGBoostConfig(BackendConfig):
     def train_func_context(self):
         @contextmanager
         def collective_communication_context():
-            with CommunicatorContext():
+            with CommunicatorContext(**_get_xgboost_args()):
```

Comment:
Are we able to save the xgboost_args into XGBoost config so we can avoid modifying the global variable?

Reply:
Hmm, interesting. I actually don't understand why we need both `BackendConfig` and `Backend` classes. Any context here @matthewdeng ?

Reply:
Yeah currently the dependency between `BackendConfig` and `Backend` are unidirectional. It's kind of hard to pass information from `Backed -> BackendConfig`.

Reply:
Should `train_func_context` be part of the `Backend` instead?

Reply:
Or at the very least the default one.

Reply:
Oh hm maybe that won't work because we construct the train loop before the backend...